include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,346 @@
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template< eve::product_type Type
+          , typename Allocator = eve::aligned_allocator<unsigned char, eve::fixed<64>>
+          >
+  struct soa_vector
+  {
+    public:
+
+    //! Type of data stored in the container
+    using value_type = Type;
+
+    using storage_type = detail::soa_storage<Type,Allocator>;
+
+    //!=============================================================================================
+    //! @name Pointers and Iterators
+    //! @{
+    //==============================================================================================
+    //!   @brief pointer and iterator types.
+    //!   pointer*  - are eve::algo::views::zip_iterator over fields (no conversion to type, just flat fields)
+    //!   iterator* - are a pointer, converter to the Type.
+    //!   They all satisfy eve::algo::relaxed_iterator but not std::iterator
+
+    using pointer               = decltype(storage_type{}.data());
+    using const_pointer         = decltype(std::declval<storage_type const>().data());
+    using pointer_aligned       = decltype(storage_type{}.data_aligned());
+    using const_pointer_aligned = decltype(std::declval<storage_type const>().data_aligned());
+
+    using iterator               = decltype(views::convert(pointer{}              , as<value_type>{}));
+    using const_iterator         = decltype(views::convert(const_pointer{}        , as<value_type>{}));
+    using iterator_aligned       = decltype(views::convert(pointer_aligned{}      , as<value_type>{}));
+    using const_iterator_aligned = decltype(views::convert(const_pointer_aligned{}, as<value_type>{}));
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    soa_vector() : storage(0) {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`. No copies are made.
+    soa_vector ( std::size_t n) : storage(n) {}
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type value)  : storage(n, value) {}
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l ) : storage(l.size())
```

Comment:
explicit please.

There are a lot of methods missing. DO we add them later?

Reply:
You tell me. We can merge that once it is OK and add other methods later or we can push forward and add them.

Reply:
No explicit agreed.
We can add the methods later. However 0 init and then fill seems annoying. does the memset from constructor actually stays there? I bet it does.